### PR TITLE
#[UIC-1765] Enabled WalkMe on RVF with feature on and off capability

### PR DESCRIPTION
**Pull request by bipinsoniguavus**
UIC-1765

### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Enabled WalkMe on RVF with feature on and off capability

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
NA

### TEST PLAN
NA

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@arpit-agarwal @ankursinghal2005 @jitendra-kumawat

### CODE REVIEW

File: superset/templates/superset/base.html [hunk 1]
```diff
@@ -18,14 +18,18 @@
...
     <link rel="icon" type="image/png" href="{{ url_for('static' , filename='assets/images/guavus.ico') }}">
-    <input type="hidden" id="app_prefix" name="app_prefix" value="{{ serverRoot }}">
     {% for entry in get_unloaded_chunks(css_manifest('theme'), loaded_chunks) %}
-      <link rel="stylesheet" type="text/css" href="{{  url_for('static' , filename='assets/dist/'+entry) }}" />
+    <link rel="stylesheet" type="text/css" href="{{  url_for('static' , filename='assets/dist/'+entry) }}" />
     {% endfor %}
+    <input type="hidden" id="app_prefix" name="app_prefix" value="{{ serverRoot }}">
+    {% if walkMeEnabled %}
+    <script type="text/javascript">(function() {var walkme = document.createElement('script'); walkme.type = 'text/javascript'; walkme.async = true; walkme.src = 'https://cdn.walkme.com/users/fe8e118680a14f439c78916022601810/test/walkme_fe8e118680a14f439c78916022601810_https.js'; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(walkme, s); window._walkmeConfig = {smartLoad:true}; })();</script>
+    {% endif %}
```
Comment (jitendra-kumawat) [inline superset/templates/superset/base.html]: can we pick this npm dependency instead hardcoding of .js path
 https://cdn.walkme.com/users/fe8e118680a14f439c78916022601810/test/walkme_fe8e118680a14f439c78916022601810_https.js

File: superset/templates/superset/basic.html [hunk 2]
```diff
@@ -107,5 +108,8 @@ <h4 class="modal-title"></h4>
         {% endwith %}
       {% endif %}
     {% endblock %}
+    {% if walkMeEnabled %}
+    <script type="text/javascript">(function() {var walkme = document.createElement('script'); walkme.type = 'text/javascript'; walkme.async = true; walkme.src = 'https://cdn.walkme.com/users/fe8e118680a14f439c78916022601810/test/walkme_fe8e118680a14f439c78916022601810_https.js'; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(walkme, s); window._walkmeConfig = {smartLoad:true}; })();</script>
+    {% endif %}
```
Comment (jitendra-kumawat) [inline superset/templates/superset/basic.html]: js link comment same as before

File: superset/templates/superset/base.html [hunk 3]
```diff
@@ -18,14 +18,18 @@
 #}
 {% extends "appbuilder/baselayout.html" %}
 {% set serverRoot =  request.script_root %}
+{% set walkMeEnabled =  request.environ['WALKME_ENABLED'] %}
```
Comment (jitendra-kumawat) [inline superset/templates/superset/base.html]: we can pick this variable directly from config like--
{% set walkMeEnabled = appbuilder.app.config.get('WALKME_ENABLED') %}

and we can avoid to set it in PrefixMiddleware env variable

File: docker/superset_config.py [hunk 4]
```diff
@@ -45,6 +45,9 @@ def get_cache_config():
 # Application Root configuration
 APPLICATION_PREFIX = get_env_variable('APPLICATION_PREFIX')
 
+# WalkMe feature enabled/disbaled
+WALKME_ENABLED = get_env_variable('WALKME_ENABLED')
+
```
Comment (jitendra-kumawat) [inline docker/superset_config.py]: use boolify before  get_env_variable to convert it in boolean

File: superset-installer/etc/reflex-provisioner/inventory/templates/group_vars/global/all/raf/superset.yml [hunk 5]
```diff
@@ -125,4 +125,6 @@ RVF_HELP_ENABLED: False
 SUPERSET_SESSION_LIFETIME_SECONDS: "3600" # stale session timeout
 RVF_LOG_LEVEL: "INFO"
 RVF_CACHE_DEFAULT_TIMEOUT: "{{ 60 * 60 * 24 }}"
-RVF_CACHE_TYPE: 'null'
\ No newline at end of file
+RVF_CACHE_TYPE: 'null'
+
+RVF_WALKME_ENABLED: True
```
Comment (jitendra-kumawat) [inline superset-installer/etc/reflex-provisioner/inventory/templates/group_vars/global/all/raf/superset.yml]: I think use of  RVF_WALKME_ENABLED  and WALKME_ENABLED  in superset-deployment.yaml  is missed

File: superset/templates/superset/base.html (hunk 3 above)
Comment (bipinsoniguavus) [reply superset/templates/superset/base.html]: I was just following the already mentioned way to maintain consistency :) I have assigned it in variable first and then reading it back to assign in environment variable.

File: superset/templates/superset/base.html (hunk 1 above)
Comment (bipinsoniguavus) [reply superset/templates/superset/base.html]: No dependency is available on npm for `walkme`

File: superset-installer/etc/reflex-provisioner/inventory/templates/group_vars/global/all/raf/superset.yml (hunk 5 above)
Comment (bipinsoniguavus) [reply superset-installer/etc/reflex-provisioner/inventory/templates/group_vars/global/all/raf/superset.yml]: Done

File: docker/superset_config.py (hunk 4 above)
Comment (bipinsoniguavus) [reply docker/superset_config.py]: Done

File: superset/templates/superset/basic.html (hunk 2 above)
Comment (bipinsoniguavus) [reply superset/templates/superset/basic.html]: No dependency is available on npm for `walkme`

**Pull request merged**